src/transaction.ts

```diff
@@ -898,30 +899,42 @@ export class Snapshot extends EventEmitter {
     );
 
     const {gaxOptions, json, jsonOptions, maxResumeRetries} = query;
-    const {params, paramTypes} = Snapshot.encodeParams(query);
-    const transaction: spannerClient.spanner.v1.ITransactionSelector = {};
-
-    if (this.id) {
-      transaction.id = this.id as Uint8Array;
-    } else {
-      transaction.singleUse = this._options;
-    }
-
-    delete query.gaxOptions;
-    delete query.json;
-    delete query.jsonOptions;
-    delete query.maxResumeRetries;
-    delete query.types;
-
-    const reqOpts: ExecuteSqlRequest = Object.assign(query, {
-      session: this.session.formattedName_!,
-      seqno: this._seqno++,
-      transaction,
-      params,
-      paramTypes,
-    });
+    let reqOpts;
+
+    const sanitizeRequest = () => {
+      const q = query as ExecuteSqlRequest;
```

Comment:
nit: Is this step needed? It seems `query` is anyways used below line [918](https://github.com/googleapis/nodejs-spanner/pull/1208/files#diff-f5daf90433414776e2e49f79b8c93adaR918)